pyERGM/metrics.py

```diff
@@ -197,6 +196,37 @@ def calculate_for_sample(self, networks_sample: np.ndarray | torch.Tensor):
         """
         return networks_sample.sum(axis=(0, 1)) // 2
 
+    def calculate_bootstrapped_features(self, first_halves_to_use: np.ndarray,
+                                        second_halves_to_use: np.ndarray,
+                                        first_halves_indices: np.ndarray[int], second_halves_indices: np.ndarray[int]):
+        """
+        Calculates the bootstrapped number of edges, by counting edges in the sampled subnetworks, and normalizing by
+        network size (i.e., calculating the fraction of existing edges out of all possible ones in sampled subnetworks,
+        and multiplying by the number of possible edges in the full observed network).
+        Parameters
+        ----------
+        first_halves_to_use
+            Multiple samples of subnetworks of an observed network, representing the connectivity between half of the
+            nodes in the large network.
+        second_halves_to_use
+            The subnetworks formed by the complementary set of nodes of the large network for each sample.
+        first_halves_indices
+            The indices of the nodes in the first half of the large network for each sample, according to the ordering
+            of the nodes in the large network.
+        second_halves_indices
+            The indices of the nodes in the second half of the large network for each sample, according to the ordering
+
+        Returns
+        -------
+        Properly normalized statistics of subnetworks of an observed network.
+        """
+        num_nodes_in_observed = first_halves_indices.shape[0] + second_halves_indices.shape[0]
+        num_nodes_in_first_half = first_halves_indices.shape[0]
+        num_possible_edges_in_observed = num_nodes_in_observed * (num_nodes_in_observed - 1) / 2
+        num_possible_edges_in_first_half = num_nodes_in_first_half * (num_nodes_in_first_half - 1) / 2
+        return self.calculate_for_sample(
```

Comment:
Why aren't we also accounting for the second half?


e.g. - 

0.5*(num_in_first / num_possible_in_first) * num_possible_in_observed + 0.5*(num_in_second / num_possible_in_second) * num_possible_in_observed